pymc4/random_variables/continuous.py

```diff
@@ -67,7 +67,7 @@ class Beta(RandomVariable):
     - beta: concentration1
     """
 
-    def _base_dist(self, alpha, beta, *args, **kwargs):
+    def _base_dist(self, alpha: float, beta: float, *args, **kwargs):
```

Comment:
Why so constraining typehints here? Pycharm will always complain

Reply:
@ferrine we can expand it! :smile: What would be the appropriate types here? I'm thinking float, np.ndarray and tf.tensor?

Reply:
Do we even need them?

Reply:
It would help with code understanding and debugging as the library would get more complex

Reply:
> It would help with code understanding and debugging as the library would get more complex

Perhaps. But because there are so many valid inputs (like float, [float], [[float1,] ,[float2]], np.array/tf.Tensor ), I'd suggest only the basic numeric types: float or int.

Keep in mind that there could be validation in the TFP Distribution.

Reply:
Leaving this as a reminder for myself: I think I can do a thing where using the Typing module, we can create a new Type hint with an alias of some kind. 

```
Vector = List[float]
```

Could do something like:

```
TensorLike = Union[List[float], float, np.ndarray, tf.Tensor]
```

That would shorten the type hint and still be informative.

Reply:
Alternatively we can force it to be tf.Tensor - as far as I understand, every will eventually convert to tf tensor in the execution at somepoint.

Reply:
> > It would help with code understanding and debugging as the library would get more complex
> 
> Perhaps. But because there are so many valid inputs (like float, [float], [[float1,] ,[float2]], np.array/tf.Tensor ), I'd suggest only the basic numeric types: float or int.
> 
> Keep in mind that there could be validation in the TFP Distribution.

The typing doesn't add anything for validation during runtime, it makes the code easier to read by humans and statically analyze by tools like mypy or pycharm

Reply:
If `tf.Tensor` will work just fine everywhere we should just use that. If not then I like @ericmjl proposal.

Reply:
Because the type hints are for end-users and developers, and not for actual type checking, I think I will implement the TensorLike type hint.